Deep flat

IrinaSing opened the pull request:
## Behavior

> What the function does - inputs and outputs

- **The Function**
  - [x] The function's name is clear and helpful
  - [x] The parameter names are clear and helpful
  - [x] Default parameters are used if possible
- **JSDoc Comment**
  - [x] The docstring is well-formatted
  - [x] There are no spelling mistakes
  - [x] The description of the function's behavior is clear
  - [x] All parameters are documented with types
  - [x] Documented parameter names match the function's parameter names
  - [x] Parameters have a good description
  - [x] If default parameters are used, they are the correct type
  - [x] The return value is documented with a type
  - [x] The return value has a clear description
- **README**
  - [x] The documentation is rendered into the README
  - [x] Docs were generated just before pushing
  - [x] Docs in the README matches the JSDoc comment
  - [x] The README contains at least one use case
- **Unit Tests**
  - [x] Unit tests are present
  - [x] There are at least 8 test cases
  - [x] All `describe`s and `it`s have clear and helpful names
  - [x] All test names actually describe what is being tested
  - [x] Tests are grouped logically using `describe`
  - [x] There is only one `expect` inside every `it`
  - [x] There is no logic (loops, conditionals, ...) in the test cases
  - [x] Side-effects are tested, if necessary
  - [x] Tests were run just before pushing
  - [x] All tests pass (`file-name.pass.txt`)
  - [x] Tricky edge cases are tested (extra credit)

## Strategy

> The logic behind the solution

- **The Function**
  - [x] Variables have names that help understand the strategy
  - [x] Spacing and formatting help to understand the strategy
  - [x] The code is written as simply and clearly as possible
  - [x] Comments _are_ used to label important goals in your strategy
  - [x] Comments _are not_ used to describe how the code works
- **README**
  - [x] There is an explanation of the solution's strategy in the README
  - [x] The explanation is clear and understandable
  - [x] The explanation actually matches the function that was written

## Implementation

> The code used to write the solution

- **Formatting**
  - [x] The code was formatted just before pushing
  - [x] No errors occurred while formatting the code
- **Linting**
  - [x] There should be no linting errors or warnings in the `file-name.lint.txt` file
  - [x] Spelling should be correct in all code, comments and docstrings
  - [x] The code was linted just before pushing
- **The solution file**
  - [x] Has a JSDoc comment
  - [x] Has an exported function
  - [x] Does no other code or comments (you can use the sandbox for saving extra ideas)
- **The function body _Should_ ...**
  - [x] Be well and consistently formatted
  - [x] Have clear and helpful variable names
  - [x] Declare variables in the smallest scope possible
  - [x] Be a simple and readable implementation
  - [x] Use short comments to describe the strategy
  - [x] Use short comments to describe important language feature choices
- **The function body should _Not_ ...**
  - [x] Have extra comments that are not about strategy or important decisions
  - [x] Have any user interactions (`prompt`, `alert`, `confirm`)
  - [x] Have any `console` calls (`.log`, `.assert`, ...)
  - [x] Have any `debugger` statements

Commit:
js, spec.js

Commit:
function, sandbox

Commit:
sandbox

Commit:
description

Commit:
linting, testing

Commit:
linting, testing

Commit:
format

IrinaSing committed:
Merge branch 'HackYourFutureBelgium:master' into deep-flat

Commit:
folder

Commit:
folder

Commit:
lint-format

Commit:
readme

Commit:
test-doc

Commit:
npm runs, linting problem fix

Commit:
name of the folder

Commit:
fresh txts

Fatihayoz commented on src/deep-flat/deep-flat-built-in-flat/README.md:
```diff
@@ -0,0 +1,78 @@
+const forBob = [["bananas", "melon"], ["tea"]];
+const foeJane = [["tomatoes", "cucumber"], ["cola"], ["snack"]];
+const forMary = [["milk"], ["apples"]];
+const shopList = [forBob, foeJane, forMary];
```
typo in the second element in the array:
"foeJane"

Fatihayoz replied:
just a little typo: foeJane --> forJane

Fatihayoz commented on src/deep-flat/deep-flat-built-in-flat/README.md:
````diff
@@ -0,0 +1,78 @@
+
+```js
+const forBob = [["bananas", "melon"], ["tea"]];
+const foeJane = [["tomatoes", "cucumber"], ["cola"], ["snack"]];
````
just a little typo: foeJane --> forJane

Fatihayoz committed:
Update README.md

Typo in line 59 and 61 corrected:
line 59: const foeJane --> const forJane
line 61: const shopList = [forBob, foeJane, forMary] --> const shopList = [forBob, forJane, forMary];

Fatihayoz reviewed (approved):
There was a very very little typo which doesn't effect the code. 
:+1: :100:
